packages/core/src/sync/index.ts

```diff
@@ -679,6 +679,22 @@ export const createSync = async (args: CreateSyncParameters): Promise<Sync> => {
                   msg: `Indexed ${events.length} events`,
                 });
               }
+
+              // update `ponder_realtime_latency` metric
+              for (const network of args.networks) {
+                for (const { block, endClock } of perNetworkSync.get(network)!
+                  .unfinalizedBlocks) {
+                  const checkpoint = encodeCheckpoint(
+                    blockToCheckpoint(block, network.chainId, "up"),
+                  );
+                  if (checkpoint > from && checkpoint <= to && endClock) {
```

Comment:
So this means, if the checkpoint for this block is within the checkpoint range that we just finished indexing, record the duration measurement?

Reply:
Yes, meaning that the block has been fully processed